CLI/Import - Test import en repos

rplevka opened the pull request:
- Got rid of redundant code (for preparing the data)
- `csv_to_dataset()` moved to `Import` class (separate commit)
- added tests for testing `hammer import repository-enable`
- manifest import tests adjusted to reflect BZ#1260722

```bash
$ nosetests test_import.py
.............................
----------------------------------------------------------------------
Ran 29 tests in 4454.522s

OK
```

rplevka commented on tests/foreman/cli/test_import.py:
```diff
@@ -27,33 +27,6 @@
 from robottelo.test import CLITestCase
 
 
-def csv_to_dataset(csv_files):
```
moved to `Import` class

rplevka commented on tests/foreman/cli/test_import.py:
```diff
@@ -299,37 +361,19 @@ def test_import_orgs_manifests(self, test_data):
-            test_data,
-            self.default_dataset[0]
-        )
-        csv_records = csv_to_dataset([files['users']])
```
this chunk of code  has been replaced by the Import static method - `organization_with_tr_data_manifests`
as it's needed in multiple tests.

rplevka commented on tests/foreman/cli/test_import.py:
```diff
@@ -222,18 +259,47 @@ def gen_import_repo_data():
                 u'channel_name': gen_string('alphanumeric'),
-                u'channel_label': gen_string('alphanumeric'),
-            } for i in range(3)]
+            } for i in set(range(101, 113)) - set(range(106, 107))] + [
```
This piece will change name of all repos excepting 107 (rhn-tools-rhel-x86_64-server-7). This will make hammer to skip them and to only recognize and synchronize a single RH repo in order to speed up the test.
rhn-tools-rhel-x86_64-server-7 is the smallest one

Ichimonji10 replied:
How about replacing `set(range(106, 107))` with `{106}`?

elyezer commented on robottelo/cli/import_.py:
```diff
@@ -36,13 +36,49 @@
 class Import(Base):
     """Imports configurations from another Satellite instances."""
     command_base = 'import'
+    repos = {
```
I think we have most of the values here in constants, I think work double checking and using the values from there.

rplevka replied:
@elyezer:
yes, i am aware of the repos defined in `constants.py`. But these are of different format. Hammer import transitions them.

sthirugn replied:
I agree with @rplevka.  Standardizing these constants should be a separate effort.  They are not really in a standard format to be consumed by all tests. They are built in format for specific tests.

elyezer replied:
:+1:

Ichimonji10 commented:
Is there any reason for this PR to be split in to two commits? It's hard for me to get insight into what's being done in these commits, as both have one-line-summary one-line-summary messages, and both affect `robottelo/cli/import.py` and `tests/robottelo/cli/test_import.py`. This will make it hard to revert changes if the need arises. If you can give more thorough commit messages, that'll help me immensely.

rplevka commented:
@Ichimonji10,
> Is there any reason for this PR to be split in to two commits?

Yes. Each commit addresses logically different change.

> It's hard for me to get insight into what's being done in these commits, as both have one-line-summary one-line-summary messages

I am very sorry to  hear that, anyway:
```bash
commit 3f6ded88e9ae279eba51d49d5ee11127a93a3a51
Author: Roman Plevka <rplevka@redhat.com>
Date:   Mon Sep 7 13:09:17 2015 +0200

    Import - csv_to_dataset moved to Import as static method
```
...which describes exactly what was done in that commit (removed a method from `test_import.py` and appended it in `import_.py`). That's why both files are affected.
I've also submitted the PR with this in the message:
> - csv_to_dataset() moved to Import class (separate commit)

I guess the only thing missing in the 2nd commit message is the fact that I created a class method for importing/enabling the RH repos and  getting rid of redundant code besides adding the tests (willl append this piece of info to it ASAP).

So thinking about it again, there should have been 3 commits in this PR :)

Ichimonji10 commented:
>> It's hard for me to get insight into what's being done in these commits, as both have one-line-summary one-line-summary messages
>
> I am very sorry to  hear that, anyway:
> ```bash
> commit 3f6ded88e9ae279eba51d49d5ee11127a93a3a51
> Author: Roman Plevka <rplevka@redhat.com>
> Date:   Mon Sep 7 13:09:17 2015 +0200
>
>     Import - csv_to_dataset moved to Import as static method
> ```
> ...which describes exactly what was done in that commit (removed a method from `test_import.py` and appended it in `import_.py`). That's why both files are affected.

No, the commit message doesn't describe "exactly what was done in that commit". I can hardly make head or tail of that summary. Also, the commit definitely does a lot more than just moving a `csv_to_dataset` method from one module to another. For example:

* Methods `organization_with_tr_data_manifests` and `repository_enable_with_tr_data` have been added to `import_.py`. Neither of these methods have been moved from `test_import.py` - they're brand new.
* Method `gen_import_rh_repo_data` has been added to `test_import.py`.
* I have no idea why any of these changes are being made. Is this related to some GitHub issue?

rplevka commented:
@Ichimonji10 
>
   - Methods organization_with_tr_data_manifests and repository_enable_with_tr_data have been added to import_.py. Neither of these methods have been moved from test_import.py - they're brand new.
   - Method gen_import_rh_repo_data has been added to test_import.py.
   - I have no idea why any of these changes are being made. Is this related to some GitHub issue?

ah! sorry. Those were supposed to land in the other commit. They are all related to test coverage of `test_import_enable_rh_repos`. I guess amended too much.

Ichimonji10 commented:
> ah! sorry. Those were supposed to land in the other commit. They are all related to test coverage of `test_import_enable_rh_repos`.

That makes more sense. Thanks for clarifying.

rplevka commented:
@Ichimonji10 : ok, I have reorganized the commits, (ended up with 3 of them) and everything sould be at the correct place + addressed the comments.

Thanks for patience.

Ichimonji10 commented:
@rplevka I'm very happy with the new commit messages. They do a great job of summarizing what changes you've made. Thank you!

Ichimonji10 commented on tests/foreman/cli/test_import.py:
```diff
@@ -222,18 +251,47 @@ def gen_import_repo_data():
-                u'organization': gen_string('alphanumeric')
+                u'organization_id': org_ids[i],
+                u'organization': gen_string('alphanumeric'),
             } for i in range(3)],
```
How about: `range(len(org_ids))`

Ichimonji10 commented on tests/foreman/cli/test_import.py:
```diff
@@ -222,18 +251,47 @@ def gen_import_repo_data():
+                    u'key': u'org_id',
+                    u'key_id': type(u'')(i + 1),
+                    u'org_id': org_ids[i],
+                } for i in range(3)
```
How about: `range(len(org_ids))`

Ichimonji10 commented:
The first commit message says:

> * Add method `organization_with_tr_data`.
> * Make method `tr_data` raise an exception if an error is encountered while
>   fetching data.

But in fact:

* The new method is named `organization_with_tr_data_manifests`.
* Several methods now raise an `AssertionError` if an error is encountered while fetching data.

Can we:

* [ ] Update the commit message to reflect these two discrepancies, and
* [ ] Update the docstrings for the methods that now raise an `AssertionError`? Something like `:raises: ``AssertionError`` if a non-zero return code is encountered.

Ichimonji10 commented:
ACK pending comments.

If this is high priority, it can be merged right away.

rplevka commented:
@Ichimonji10 
comments addressed

Ichimonji10 commented:
Thank you @rplevka. ACK.

sthirugn commented:
ACK
